mmdet/apis/train.py

```diff
@@ -213,6 +213,17 @@ def build_optimizer(model, optimizer_cfg):
         return optimizer_cls(params, **optimizer_cfg)
 
 
+def add_logging_on_first_and_last_iter(runner):
+    def every_n_inner_iters(self, runner, n):
+        if runner.inner_iter == 0 or runner.inner_iter == runner._max_iters - 1:
+            return True
+        return (runner.inner_iter + 1) % n == 0 if n > 0 else False
+
+    for i, _ in enumerate(runner.hooks):
```

Comment:
What's the reason to iterate over indices, but not the values themselves?

Reply:
sorry about that, fixed